tests/__init__.py

```diff
@@ -359,3 +361,58 @@ def assert_url_equal(url1, url2):
     assert_equal(parts1.hostname, parts2.hostname)
     assert_equal(parts1.port, parts2.port)
     assert_equal(parse_qs(parts1.query), parse_qs(parts2.query))
+
+
+class HTTPStubberException(Exception):
+    pass
+
+
+class RawResponse(BytesIO):
+    def stream(self, **kwargs):
+        contents = self.read()
+        while contents:
+            yield contents
+            contents = self.read()
+
+
+class ClientHTTPStubber(object):
+    def __init__(self, client):
+        self.reset()
+        self._client = client
+
+    def reset(self):
+        self.requests = []
```

Comment:
As a general rule, I would agree. Though in this case, I want `reset` to basically do what `__init__` does so it makes sense to put all of the initialization logic in `reset` and just call it from `__init__`. I don't feel too strongly on this, mostly did it to remove duplication and ensure `reset` gets you back to a clean slate in case this class ever got more complicated.

Reply:
I'm -0 on keeping it since it doesn't actually violate a PEP